File: sov_school_back/back_common/src/main/java/com/sovereingschool/back_common/Models/Curso.java
package com.sovereingschool.back_common.Models;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name = "curso")
public class Curso implements Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id_curso")
	private Long idCurso;

	@Column(name = "nombre_curso", unique = true, nullable = false)
	private String nombreCurso;

	@ManyToMany(fetch = FetchType.EAGER)
	@JoinTable(name = "curso_profesor", joinColumns = @JoinColumn(name = "id_curso"), inverseJoinColumns = @JoinColumn(name = "id_usuario"))
	@JsonIgnoreProperties({ "rollUsuario", "planUsuario", "cursosUsuario", "fechaRegistroUsuario" })
	private List<Usuario> profesoresCurso;

	private Date fechaPublicacionCurso;

	@OneToMany(mappedBy = "cursoClase", fetch = FetchType.EAGER, cascade = CascadeType.ALL, orphanRemoval = true)
	@JsonManagedReference
	@JsonIgnoreProperties({ "cursoClase" }) // evita recursión infinita
	private List<Clase> clasesCurso;

	@ManyToMany(fetch = FetchType.EAGER)
	@JoinTable(name = "cursos_plan", joinColumns = @JoinColumn(name = "id_curso"), inverseJoinColumns = @JoinColumn(name = "id_plan"))
	@JsonIgnoreProperties({ "precioPlan", "cursosPlan" })
	private List<Plan> planesCurso;

	private String descripcionCorta;

	@Column(length = 1500)
	private String descripcionLarga;

	private String imagenCurso;

	private BigDecimal precioCurso;

}

File: front/src/app/models/Curso.ts
import { Clase } from './Clase';

export class Curso {
	public idCurso: number;

	public nombreCurso: string;

	public profesoresCurso: number[];

	public fechaPublicacionCurso?: Date;

	public clasesCurso?: Clase[];

	public planesCurso?: number[];

	public descripcionCorta: string;

	public descripcionLarga?: string;

	public imagenCurso: string;

	public precioCurso?: number;

	constructor(_idCurso: number, _nombreCurso: string, _profesoresCurso: number[], _descripcionCorta: string, _imagenCurso: string, _fechaPublicacionCurso?: Date, _clasesCurso?: Clase[], _planesCurso?: number[], _descripcionLarga?: string, _precioCurso?: number) {
		this.idCurso = _idCurso;
		this.nombreCurso = _nombreCurso;
		this.profesoresCurso = _profesoresCurso;
		this.fechaPublicacionCurso = _fechaPublicacionCurso;
		this.clasesCurso = _clasesCurso;
		this.planesCurso = _planesCurso;
		this.descripcionCorta = _descripcionCorta;
		this.descripcionLarga = _descripcionLarga;
		this.imagenCurso = _imagenCurso;
		this.precioCurso = _precioCurso;
	}
}
